functions/go/set-setter/main.go

```diff
@@ -0,0 +1,160 @@
+// This file will be processed and embedded to pluginator.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-annotation/setters"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/wrappy"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+	"sigs.k8s.io/yaml"
+)
+
+type setSetterSpecs struct {
+	Setters []setSetterSpec `json:"setters,omitempty" yaml:"setters,omitempty"`
+}
+
+type setSetterSpec struct {
+	SetterName  string      `json:"setter_name,omitempty" yaml:"setter_name,omitempty"`
+	SetterValue interface{} `json:"setter_value,omitempty" yaml:"setter_value,omitempty"`
+}
+
+func setSetters(specs setSetterSpecs,
+	resMap resmap.ResMap,
+	pluginHelpers *resmap.PluginHelpers,
+	plugin *plugin) error {
+
+	err := plugin.Config(pluginHelpers, []byte{})
+	if err != nil {
+		return fmt.Errorf("failed to config plugin: %w", err)
+	}
+	// set setter key and value
+	for _, spec := range specs.Setters {
+		plugin.Setters = append(plugin.Setters, setters.Setter{
+			SetterName:  spec.SetterName,
+			SetterValue: spec.SetterValue,
+		})
+	}
+
+	err = plugin.Transform(resMap)
+	if err != nil {
+		return fmt.Errorf("failed to run transformer: %w", err)
+	}
+	return nil
+}
+
+//nolint
+func main() {
+	var plugin *plugin = &KustomizePlugin
+	resmapFactory := newResMapFactory()
+
+	pluginHelpers := newPluginHelpers(resmapFactory)
+
+	resourceList := &framework.ResourceList{}
+	resourceList.FunctionConfig = map[string]interface{}{}
+
+	cmd := framework.Command(resourceList, func() error {
+		resMap, err := resmapFactory.NewResMapFromRNodeSlice(resourceList.Items)
+		if err != nil {
+			return err
+		}
+		setters, err := getSetters(resourceList.FunctionConfig)
+		if err != nil {
+			return fmt.Errorf("failed to get data.specs field from function config: %w", err)
+		}
+		err = setSetters(setters, resMap, pluginHelpers, plugin)
+		if err != nil {
+			return fmt.Errorf("failed to set setters: %w", err)
+		}
+
+		resourceList.Items, err = resMap.ToRNodeSlice()
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	cmd.Long = usage()
+	if err := cmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func usage() string {
+	return `Set values of resources with setter values.
+
+Configured using a ConfigMap with the following keys:
+
+setter_name: Setter name.
+setter_value: Setter value.
+
+These keys are in a list in path 'data.setters'.
+
+Example:
+
+To set a setter 'color: orange' to all resources:
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  setters:
```

Comment:
what's the purpose of having `setters` wrapper struct? I can see having `setterValues` potentially be useful if we want to extend the function config in the future with additional metadata:

setterValues:
  foo: bar

Reply:
Tried to have it consistent with other functions.
https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/set-annotation/main.go#L135